### PR TITLE
Hide current screen

### PR DESCRIPTION
**Pull request by via-my**
What about adding an easier way to hide current screen? This is really efficient for projects that use a lot of them.

### CODE REVIEW

**Comment by renpytom** [comment]
I think I like the idea of it, not 100% sold on the name, as self is used a lot in Python. Perhaps HideCurrent() ?

**Comment by Gouvernathor** [comment]
What does this do when screen A `use`s screen B, thelatter including a button with this action on it ? Does it rightfully hide screen A ?

Also, about the name, how about making this the behavior of the Hide action when no arguments is passed to it ? Which it doesn't currently support.

**Comment by renpytom** [comment]
It would hide screen A, which is the right behavior. Hide with no arguments could work as well. @via-my, what do you think about that option?

**Comment by via-my** [comment]
Sounds much better :)
Meh, it looks a little bit messy... Sorry.

File: renpy/common/00action_control.rpy [hunk 1]
```diff
@@ -201,16 +202,24 @@ init -1500 python:
...
             self._layer = _layer
 
         def __call__(self):
+            if self.screen is None:
+                cs = renpy.current_screen()
+
+                if cs is None:
+                    return
+
+                self.screen = cs.screen_name
```
Comment (mal) [inline renpy/common/00action_control.rpy:217]: In order to work with screens that declare a preferred `layer`, and also those called with `_layer=`, I think we also need to set the layer from the current screen before calling `hide_screen`.
```suggestion
                self.screen = cs.screen_name
                self._layer = cs.layer
```

Comment (via-my) [reply renpy/common/00action_control.rpy:217]: Well spotted!

**Comment by Andykl** [comment]
I do not like the idea that `__call__` mutates the object itself.
I can have such code
```
define hide_current_screen = [Hide(), With(Dissolve(0.5), maybe some other actions]

screen some1():
    timer 1.0 action hide_current_screen

screen some2():
    timer 1.0 action hide_current_screen
```

And your code after first use with screen some1 will not work with screen some2.

The solution is easy, just compute the screen and layer as local variables instead of setting self.screen in `__init__`.

**Comment by via-my** [comment]
I guess this should work now. Thank you!

File: renpy/common/00action_control.rpy [hunk 2]
```diff
@@ -191,26 +191,37 @@ init -1500 python:
...
+
+            else:
+                renpy.hide_screen(self.screen, layer=self._layer)
 
             if self.transition is not None:
                 renpy.transition(self.transition)
 
             renpy.restart_interaction()
-
+
```
Comment (Andykl) [inline renpy/common/00action_control.rpy:227]: Spurious spaces

**Review by Andykl — APPROVED** [review]
Nit thing. Except for that LGTM.

File: renpy/common/00action_control.rpy [hunk 3]
```diff
@@ -191,26 +191,37 @@ init -1500 python:
...
-        def __init__(self, screen, transition=None, _layer=None):
+        def __init__(self, screen=None, transition=None, _layer=None):
             self.screen = screen
             self.transition = transition
             self._layer = _layer
 
         def __call__(self):
-            renpy.hide_screen(self.screen, layer=self._layer)
+            if self.screen is None:
+                cs = renpy.current_screen()
```
Comment (Gouvernathor) [inline renpy/common/00action_control.rpy:213]: This function does not look documented. If I have two (independant) screens being displayed, A and B, how is it supposed to tell the two apart ?

Comment (via-my) [reply renpy/common/00action_control.rpy:213]: Seems that renpy.current_screen() only works in the context of screen, so it shouldn't be misleading.

Comment (Gouvernathor) [reply renpy/common/00action_control.rpy:213]: Ok, I tested for myself and I don't know how, but it works.

**Review by Gouvernathor — APPROVED** [review]

**Comment by renpytom** [comment]
Merged, thanks. I made minor fixes in 90d5ee811.

**Pull request merged**